nomad/local/deploy_generator.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+# Creates a namespace in nomad and deploys a generator into it
+
+read -p "Tenant ID: " tenant_id
+read -p "Generator Ingest: " generator_ingest
+read -p "Artifact URL: " artifact_url
+read -p "Instances: " generator_count
```

Comment:
How is this script meant to be invoked? Why are you using `read` rather than just referencing arguments?

Reply:
I made this script as Ian had brought up concerns about the difficulty of local development. I'm far from a bash expert so, unfortunately, I don't really know of the alternative way to do this. 

If you have an alternative recommendation, would you mind showing me what you mean by referencing arguments?

Reply:
``` bash
#!/usr/bin/env bash

set -euo pipefail

readonly tenant_id="${1}" # the first argument
readonly generator_ingest="${2}" # the second argument, etc.
readonly artifact_url="${3}"
readonly generator_count="${4}"

# rest of script follows
```
You can get more fancy with flags and whatnot, but this would be the minimum.

This would let you call it like so:

``` bash
nomad/local/deploy_generator.sh my-tenant-id sysmon https://blah.com/artifact.tar.gz 1
```
As it exists now, you can only use this script interactively, which limits it's usefulness.

Incidentally, there is also nothing in the current script that prevents you from just mashing Enter at all those prompts and having a lot of empty variables, which will end in tears. With the `set -u` flag in my example above, it will at least require that all those arguments are present.

Reply:
i mean the easiest thing (which I personally don't like) is
```
readonly tenant_id="$1"
readonly generator_ingest="$2"
```

or, more of a pain but more explicit:
```
TENANT_ID=12345 GENRATOR_INGEST=12345 deploy_generator.sh
```

Reply:
Ah! I see what you mean now. I choose interactively because I hate having a billion command line arguments if I can help it, but this is totally fine too. I'll update it to use command line arguments instead. Thanks for the help :)

Reply:
If you _really_ want interactive, you can always add an `--interactive` option, that (if present) would use this `read`-based logic. If it's not there, then it assumes you're passing everything in as arguments.

Reply:
It's fine; I'm not married to the idea. I can just use command line arguments hahaa